File: GeeksForGeeks/Basic/Encrypt_the_string_1.java
import java.util.Scanner;

class GFG
{
  public static String reverse(String input)
  {
      String output = "";
      for(int i = 0; i < input.length(); ++i)
      {
          output = input.charAt(i) + output;
      }
      return output;
  }
	public static void main (String[] args)
	{
	    Scanner scan = new Scanner(System.in);
	    int testcase = scan.nextInt();
	    char testCharacter;
      int count;
      String str;
      String result;
      scan.nextLine();
    	while(testcase != 0)
      {
          result = "";
          str = scan.nextLine();
          testCharacter = str.charAt(0);
          count = 1;
          for(int i = 1; i < str.length() && str.charAt(i) != ' '; ++i)
          {
              if (str.charAt(i) == testCharacter)
              {
                  count += 1;
              }
              else
              {
                  result += testCharacter + "" + count;
                  testCharacter = str.charAt(i);
                  count = 1;
              }
          }
          result += testCharacter + "" + count;
    	    System.out.println(reverse(result));
          --testcase;
    	}
    	scan.close();
	}
}

File: GeeksForGeeks/Basic/Encrypt_the_string_1.cpp
#include <iostream>
#include <string>
using namespace std;

string reverse(string input)
{
    string output = "";
    for(int i = 0; i < input.length(); ++i)
    {
        output = input[i] + output;
    }
    return output;
}

int main()
{
    int test_case;
    ios::sync_with_stdio(false);
    cin >> test_case;
    string str, result;
    while (test_case--)
    {
        cin >> str;
        char test_character = str[0];
        int count = 1;
        result = "";
        int j = 0;
        for(int i = 1; i < str.length(); ++i)
        {
            if (test_character == str[i]) {
                count += 1;
            }
            else
            {
                result += test_character;
                result += to_string(count);
                test_character = str[i];
                count = 1;
            }
        }
        result += test_character + to_string(count);
        cout << reverse(result) << endl;
    }
}
